appcli/variables_manager.py

```diff
@@ -43,86 +55,198 @@ def get_variable(self, path):
 
+        Throws:
+            Exception: Unable to read the configuration file.
+
+        Returns:
+            Dict: The configuration data from the main configuration file.
+        """
+        config_variables_main = dict()
+        try:
+            data_string = self.configuration_file.read_text(encoding="utf-8")
         except Exception as ex:
             raise Exception(
-                f"Could not read configuration file at [{self.configuration_file}]"
+                f"Could not read main configuration file at [{self.configuration_file}]"
             ) from ex
+        config_variables_main |= self._convert_yaml_to_dict(
+            data_string, self.configuration_file.stem
+        )
+        return config_variables_main
+
+    def _get_extra_configuration(self, variables: Dict) -> Dict:
+        """Gets the configuration from the additional configuration files.
 
-    def __save(self, variables: Dict):
+        Args:
+            variables (Dict): Variables used to populate the template(s).
+
+        Throws:
+            Exception:
+                Unable to read the configuration file.
+                Could not correctly render a jinja2 template.
+
+        Returns:
+            Dict: The configuration data from the additional configuration files.
+                Each config file is a seperate dictionary with its filename as the key.
+        """
+        if self.extra_configuration_dir is None:
+            return {}  # User didn't provide additional config directory.
+        config_variables = dict()
+        for config_file in self.extra_configuration_dir.glob("*"):
+            try:
+                data_string = config_file.read_text(encoding="utf-8")
+            except Exception as ex:
+                raise Exception(
+                    f"Could not read configuration file at [{config_file}]"
+                ) from ex
+            if config_file.suffix == ".j2":  # Jinja2 file.
+                try:
+                    data_string = self._convert_jinja_to_yaml(data_string, variables)
+                except Exception as ex:
+                    raise Exception(
+                        f"There was a problem rendering the jinja2 file at [{config_file}]"
+                    ) from ex
+            config_variables |= self._convert_yaml_to_dict(
+                data_string, config_file.stem
+            )
+        return config_variables
+
+    def _save(self, variables: Dict):
         """Saves the supplied Dict of variables to the configuration file
 
         Args:
-            variables (Dict): the variables to save
+            variables (Dict): All the variables to save.
         """
         full_path = self.configuration_file.absolute().as_posix()
         logger.debug(f"Saving configuration to [{full_path}] ...")
         with open(full_path, "w") as config_file:
             self.yaml.dump(variables, config_file)
+
+    def _convert_jinja_to_yaml(self, jinja_source: str, variables: Dict) -> str:
```

Comment:
This function takes in a template string, and variables, and returns the rendered template

Rename to `_render_j2` and remove references to `yaml` (since this function could technically be applied to anything)